lib/preview/createRoutes.js

```diff
@@ -6,13 +6,24 @@ const { buildModuleRouteHandler } = require('./routes/module.js');
 const { buildTemplateRouteHandler } = require('./routes/template.js');
 const { buildMetaRouteHandler } = require('./routes/meta.js');
 
+const { proxyResourceRedirectHandler } = require('./routes/proxyResourceRedirect.js');
+
 const createPreviewServerRoutes = async (sessionInfo) => {
     const previewServerRouter = Router();
-    previewServerRouter.get('/', buildIndexRouteHandler(sessionInfo));
     previewServerRouter.get('/proxy', buildProxyRouteHandler(sessionInfo));
     previewServerRouter.get('/module/:name', buildModuleRouteHandler(sessionInfo));
     previewServerRouter.get('/template/:name', buildTemplateRouteHandler(sessionInfo));
     previewServerRouter.get('/meta', buildMetaRouteHandler(sessionInfo));
+
+    previewServerRouter.get('/*', proxyResourceRedirectHandler);
+    previewServerRouter.post('/*', proxyResourceRedirectHandler);
+    previewServerRouter.delete('/*', proxyResourceRedirectHandler);
+    previewServerRouter.head('/*', proxyResourceRedirectHandler);
+    previewServerRouter.put('/*', proxyResourceRedirectHandler);
+    previewServerRouter.options('/*', proxyResourceRedirectHandler);
+
+    previewServerRouter.get('/', buildIndexRouteHandler(sessionInfo));
```

Comment:
Index must be moved to the bottom priority - else a relative link to `www.example.com` (`/`) goes to `localhost:3000/` and will go to that handler first. If the referrer isn't from the path `/proxy`, i.e. wasn't navigated to from a page we're proxying but rather the user just went to `localhost:3000` manually, then we `next()` and take index instead